tests/integ/conftest.py

```diff
@@ -63,26 +79,71 @@ def nonce(request):
     return request.config.getoption("--nonce")
 
 
-@pytest.yield_fixture(autouse=True)
-def cleanup_objects(engine):
-    yield
-
-    # TODO track bound models w/model_bound signal (TODO), then use boto3 to scan/delete by Meta.table_name
-    # Running tests individually may break if the User table isn't bound as part of that test
-    users = list(engine.scan(User))
-    engine.delete(*users)
-
-
 @pytest.fixture(scope="session")
-def dynamodb():
-    return fixed_ddb_client
+def dynamodb(do_provide_aws):
+    return fixed_session.client("dynamodb", endpoint_url=resource_options['endpoint_url'])
 
 
 @pytest.fixture(scope="session")
-def dynamodbstreams():
-    return fixed_streams_client
+def dynamodbstreams(do_provide_aws):
+    return fixed_session.client("dynamodbstreams", endpoint_url=resource_options['endpoint_url'])
 
 
 @pytest.fixture
 def engine(dynamodb, dynamodbstreams):
-    return Engine(dynamodb=dynamodb, dynamodbstreams=dynamodbstreams)
+    yield Engine(dynamodb=dynamodb, dynamodbstreams=dynamodbstreams)
+
+
+def get_open_port():
+    with contextlib.closing(
+            socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+        s.bind(("", 0))
+        s.listen(1)
+        port = s.getsockname()[1]
+        return port
+
+
+def get_ddb_local():
+    localdir = pytest.config.rootdir.join('.dynamodb-local').strpath
+    if not os.path.exists(localdir):
```

Comment:
Do the dynamodb docs indicate an etag that we can request to pin a version?  I'm not ok with downloading arbitrary code and executing it, even if we are confident that the publisher (dynamodb's bucket) is trustworthy.

I'll feel much better about this if either:
1. We can specify an etag to ensure the same blob is downloaded (and ideally a checksum that we can hardcode)
2. We don't download as part of the tests, but include a script that downloads and sets up the binary.

The first is preferred, since it keeps the test/dev setup simple but gives us guarantees about the code running, while the second at least ensures we know when the download occurs.

Reply:
I'll make sure it matches a sha256 hash they publish on their site.